### PR TITLE
All forks and PRs merge

### PR DESCRIPTION
**Pull request by AOx0**
This is a quick merge of all pull requests and forks.

## Download new Community Edition versions at [Digital-Logic-Sim-CE/releases](https://github.com/DigitalLogicSimCommunity/Digital-Logic-Sim-CE/releases/latest). (Current: v0.39.1)

Huge thanks to everyone that has contributed! There is still a lot of work to do on the current features and lots of options to explore. Any contributions are welcome at [DigitalLogicSimCommunity/Digital-Logic-Sim-CE](https://github.com/DigitalLogicSimCommunity/Digital-Logic-Sim-CE/)!

### CODE REVIEW

**Comment by okyloky9** [comment]
Not sure if issue from your bundle or from the author of the chip view, but just want to point out that chip viewing doesn't work completely correctly. The spawned inputs don't interact with the wires at all, similar to my previous post about why I couldn't make it work in #63

**Comment by AOx0** [comment]
It's a thing of my bundle, I'll work on it.

**Comment by AOx0** [comment]
@okyloky9. I checked it out and seems like the problem is that the View/Update feature is not compatible with saves of previous versions of DLS, specifically with chips created on previous versions of the simulator.

Another similar problem is a un-interactable input signal appearing sometimes due to the program failing to load new default chips.

This is in my Mac tho, can you try creating a new 'Project' and testing View/Update there?

**Comment by AOx0** [comment]
@okyloky9 I should work now. Any other problem let me know.

PD. Make a backup of your saves before trying v0.34.

**Comment by WafflesDev** [comment]
Since this seems to be the largest "mod" for the sim, i have some suggestions. 1. light that is default black, and white when powered. 2. keyboard input. turns on when a certain key is down, could be a special naming format like {key}.

**Comment by Turnip1234** [comment]
I don't understand what either suggestion is, could you rephrase?

**Comment by WafflesDev** [comment]
> I don't understand what either suggestion is, could you rephrase?

@Turnip1234 sorry, in my head while i was typing it, i thought it was clear. for the light, theres a new block that only has 1 input. when the input is off, the block is black, when it is on the block is white, this can be integrated with the multi bit lines to allow greys, like when inputing a 4 bit line, there are 4 shades, etc. 

With the keyboard thing, its a new input on the left of the screen,  that activates when ever the <key> is pressed down. you can create it by naming an input {<key>} where key is any ascii key

**Comment by NobreHD** [comment]
Well, I was editing some circuits when the save button got stuck in "Update"

**Comment by Turnip1234** [comment]
Yes this is a known bug @NobreHD
If you exit and reload the project it will be fixed
Sometimes, all of your chips will seem to be gone, but this is because of a file named ".txt" in your save data directory

**Comment by NobreHD** [comment]
> Sometimes, all of your chips will seem to be gone, but this is because of a file named ".txt" in your save data directory
Yeah I noticed

**Comment by Turnip1234** [comment]
If you delete the file, it will be fixed

**Comment by Turnip1234** [comment]
It's in the path `%appdata%\..\LocalLow\Sebastian Lague\Digital Logic Sim\SaveData\`
then go into the folder for your project, and delete the file name ".txt"
there'e one in the project directory, and one in the WireLayout folder in the project directory

**Comment by Turnip1234** [comment]
( on windows)

**Comment by NobreHD** [comment]
I have a suggestion: add a clock chip, as many projects I come across use clocks to keep in sync
and thanks for the quick reply

> It's in the path %appdata%\..\LocalLow\Sebastian Lague\Digital Logic Sim\SaveData\

Thank, I found it

**Comment by Turnip1234** [comment]
You can make a clock chip yourself, with logic gates @NobreHD

**Comment by Turnip1234** [comment]
image
^
@NobreHD This is an example of a toggleable clock

**Comment by NobreHD** [comment]
Thank for the help

**Comment by thetayloredman** [comment]
There is no "32-BIT DECODER", yet there being a "32-Bit Bus".

**Comment by thetayloredman** [comment]
And if you repeatedly edit a chip removing/adding busses, they will slowly move off the screen.

**Comment by thetayloredman** [comment]
But those busses are really cool, I created a 64-bit register with them finally!

**Comment by diguifi** [comment]
I personally don't like the idea of having a lot of premade stuff  on my "gates' bar" by default (those decoders and HDD).
Me, as a beginner, like to have my initial project workspace empty, just with the default "AND" and "NOT" gates implemented. I have no idea what bit decoders are and how they work, being there by default just makes my workspace feel "dirty" with a bunch of nonsense chips to me.

**Comment by Turnip1234** [comment]
@diguifi I totally agree, Digital Logic Sim was created to learn about how computers work, and to mess around with logic gates. I just had a bunch of fun adding new chips, as I'm sure other people did too. This version is also probably 1,000x more buggy than the vanilla version, I had Unity crash on me probably like 20 times. But luckily, this PR most likely will never get merged, and it's only for people who want to go further. I started looking into a way to make a separate menu for builtin chips to try and remove clutter, but it's nowhere near complete and I'm not actively working on it.
But whatever version of DLS you choose, have fun!

**Comment by diguifi** [comment]
@Turnip1234 understandable! I thought this PR meant a "new future version" rather than an experimental branch. Thanks for clarifying!

**Comment by Turnip1234** [comment]
Wooo!

**Comment by AOx0** [comment]
v0.35 out

**Comment by RedStoner** [comment]
i think it had to do with how i updated from the vanilla branch to this branch. I partially have it working now. Editing chips does not work for me. and there are some major issues with rendering that are fixed by pausing sim and resuming. Also running into issues where saved circuits just dont seem to work all the time. 
this is the save i have been working with
[TIMER.zip](https://github.com/SebLague/Digital-Logic-Sim/files/7326523/TIMER.zip)
.

**Comment by AOx0** [comment]
> are you getting same issues? @RedStoner 

Yes, I do. I created an Issue to keep talking about the problem there.

**Comment by torchsmith** [comment]
Hey @AOx0 do you think we could move this project to a new public repo so we can have more open contributors? It doesn't seem like Seb is devoting any more time on this project to look at and/or accept pull requests.

**Comment by AOx0** [comment]
@torchsmith Here it is https://github.com/AOx0/Digital-Logic-Sim.
I think I will close this PR later, then.

**Comment by avipars** [comment]
AMAZING

**Comment by AOx0** [comment]
New release (v0.36) [here](https://github.com/AOx0/Digital-Logic-Sim-CE/releases/tag/v0.36) with lots of new features by [SwissCorePy](https://github.com/SwissCorePy). New zoom feature, minimap, change chips scale, chip folder system, persistent HDD, etc. 

Make sure to backup your saves before trying.

**Comment by AOx0** [comment]
New release (v0.37) [here](https://github.com/AOx0/Digital-Logic-Sim-CE/releases/latest) with various bugs fixes and now with a Linux build.

Make sure to backup your saves before trying.

**Comment by Turnip1234** [comment]
Nice!

**Comment by Turnip1234** [comment]
This is quite nitpicky and I really don't mind, but the 8x8 display was writen by me and not Tigralt (although they might've improved it at some point, I do not know).

**Comment by Turnip1234** [comment]
On another note, this is really amazing! I love the tiny details like the loading bar!

**Comment by AOx0** [comment]
Omg. I will get that detail fixed. Sorry!!
Ps. Glad you like the current version :D

**Comment by AOx0** [comment]
New release (v0.38) at [Digital-Logic-Sim-CE/releases](https://github.com/DigitalLogicSimCommunity/Digital-Logic-Sim-CE/releases/latest) with various bugs fixes and now with a Linux build.

This new version includes:
- Fixes Chip Import for the Linux build, for the Mac Universal build and hopefully for Silicon Macs.
- There's a new chip dependency load system by [VitoBarra](https://github.com/VitoBarra)

Also moved the project to a new organization  [DigitalLogicSimCommunity](https://github.com/DigitalLogicSimCommunity)
Make sure to backup your saves before trying.
